src/coreclr/jit/morph.cpp

```diff
@@ -12211,8 +12211,11 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                     {
                         tree = gtFoldExpr(tree);
                     }
+                    else
```

Comment:
This presumes `gtFoldExpr` will actually fold, and while I think it likely will do so, I don't believe it is a guarantee. Did you notice any new diffs once this change went in?

Perhaps it's better to just retest `tree` to see if it's been folded to a constant, and call `CheckDivide...` if not.

Reply:
So, the reason that I went with this is that we are guaranteed to have two constant operands for a `UMOD` at this point, and `UsesDivideByConstOptimized` gives up in the case of a constant dividend anyway:
https://github.com/dotnet/runtime/blob/5761dd49339911241209c1ffd61105a94edb6df1/src/coreclr/jit/gentree.cpp#L6873-L6879

It does seem a bit fragile and subtle though, now that I think about it. I will rewrite it as per the suggestion.